File: CartQuestAndroid/app/src/main/java/com/amoghghadge/cartquestandroid/data/model/CartItem.kt
package com.amoghghadge.cartquestandroid.data.model

data class Substitute(
    val productId: String = "",
    val name: String = "",
    val brand: String = ""
)

data class CartItem(
    val productId: String = "",
    val name: String = "",
    val brand: String = "",
    val imageUrl: String = "",
    val quantity: Int = 1,
    val substitutes: List<Substitute> = emptyList()
)

File: CartQuestiOS/CartQuestiOS/Models/CartItem.swift
import Foundation

struct Substitute: Codable, Identifiable {
    var id: String { productId }
    var productId: String = ""
    var name: String = ""
    var brand: String = ""
}

struct CartItem: Codable, Identifiable {
    var id: String { productId }
    var productId: String = ""
    var name: String = ""
    var brand: String = ""
    var imageUrl: String = ""
    var quantity: Int = 1
    var substitutes: [Substitute] = []
}
